### PR TITLE
added recipe for bilm

### PR DESCRIPTION
**Pull request by PhilipMay**
added recipe for bilm - nothing else to say :-)

### CODE REVIEW

**Comment by PhilipMay** [comment]
Does anybody have an idea why this fails on Mac and Linux build but not on Windows? On my local Mac build everything is fine.

**Comment by marcelotrevisani** [comment]
It seems you will need the ``c`` compiler

File: recipes/bilm/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  # there is no Github release - so this ugly link is needed
+  url: https://github.com/allenai/bilm-tf/archive/ebf52c6ec1012a3672247c2d14ff7bcad7fb812b.zip
+  sha256: 5722ebd46f4120823469e1c221d4123cc6a1bc33b7083243e0893066e6f5a0f1
+
+build:
+  noarch: python
```
Comment (marcelotrevisani) [inline recipes/bilm/meta.yaml]: I think you have to remove the ``noarch`` and add a ``build`` section inside of the ``requirements`` and after that add
```
  build:
    - {{ compiler('c') }}
```

**Comment by PhilipMay** [comment]
This is a 100% pure python package. See here: https://github.com/allenai/bilm-tf
Why do I need a compiler?

**Comment by marcelotrevisani** [comment]
It seems that tensorflow needs it

**Comment by PhilipMay** [comment]
Even the setup.py file is very simpel to build the PyPI package: https://github.com/PhilipMay/bilm-tf/blob/setup.py/setup.py

**Comment by PhilipMay** [comment]
> It seems that tensorflow needs it
Yes TF needs c compiler for sure. But I do not build tensorflow... I am confused.

**Comment by marcelotrevisani** [comment]
probably TF needs the runtime libraries and it's not adding it into the package
It may be an error in the build process of TF or it's normal to add this, lets see if the build will pass now

**Comment by PhilipMay** [comment]
Well, now all builds fail. Even the Mac Build. Do you have any idears?

**Comment by marcelotrevisani** [comment]
Yes, maybe. As I don't have experience with TF I will mark @jakirkham and @dougalsutherland they may have a hint here

**Comment by jjhelmus** [comment]
The `tensorflow` package in conda-forge is a repack of the wheels provided on PyPI.  These wheels do not have the same compatibility as other packages in conda-forge.  As such, tensorflow may not import on the build machines. For example, on Linux the wheel supports Ubuntu 16.04 or later and is incompatible with the CentOS 6 docker image used for builds, see conda-forge/tensorflow-feedstock#19, tensorflow/tensorflow#8802.  Building from source, conda-forge/tensorflow-feedstock#6 may solve this issue but is challenging.

**Comment by PhilipMay** [comment]
@jjhelmus Ok - so I have to wait until the tensorflow package is fixed so my build does not fail?

**Comment by PhilipMay** [comment]
@jjhelmus Or should I just remove the tests like the tensorflow guys did:
https://github.com/conda-forge/tensorflow-feedstock/blob/034249e4e9a5a026f4bda812dfadef89ff5199b3/recipe/meta.yaml#L46

**Comment by jjhelmus** [comment]
> tensorflow is not build for python 2.7 on windows

Tensorflow is limited to Python 3.x on Windows.  If you require tensorflow at run time you will need to skip the Python 2.7 build on Windows.

**Comment by PhilipMay** [comment]
I removed the tensorflow dependency so you can decide if you want to use tensorflow-gpu or just tensorflow. Like it has been done in the setup.py of bilm: https://github.com/allenai/bilm-tf/blob/master/setup.py

**Comment by PhilipMay** [comment]
Can someone trigger a rebuild please?

**Comment by jakirkham** [comment]
Could you please merge with `master`? A lot has changed with our CI setup since this PR was opened.

File: recipes/bilm/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{% set name = "bilm" %}
+{% set version = "0.1" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  # there is no Github release - so this ugly link is needed
```
Comment (scopatz) [inline recipes/bilm/meta.yaml]: How do we know this is version 0.1 if there are no releases?

File: recipes/bilm/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  url: https://github.com/allenai/bilm-tf/archive/ebf52c6ec1012a3672247c2d14ff7bcad7fb812b.zip
+  sha256: 5722ebd46f4120823469e1c221d4123cc6a1bc33b7083243e0893066e6f5a0f1
+
+build:
+  number: 0
+  script: "{{ PYTHON }} -m pip install . --no-deps --ignore-installed --no-cache-dir -vvv"
```
Comment (scopatz) [inline recipes/bilm/meta.yaml]: Can this be `noarch: python`?

**Pull request closed without merging**